gen/src/index.ts

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env node
+const yargs = require('yargs');
+const mkdirp = require('mkdirp');
+const fs = require('fs');
+const sqrl = require('squirrelly');
+
+const qt = require('qt');
+
+let IN = yargs.argv.in || process.env.IN;
+let OUT = yargs.argv.out || process.env.OUT;
+
+const LANGUAGE = 'go';
+const TEMPLATE_DIRECTORY = 'templates';
+const GO_FILENAME = 'events.go';
+const SNIPPET_DIRECTORY = 'snippets';
+const MARSHAL_FUNCS_TO_REPLACE_SNIPPET = 'marshalFuncsToReplace.snippet';
+const MARSHAL_FUNCS_TEMPLATE = 'marshalFuncs.squirrelly';
+const README_TEMPLATE = 'README.squirrelly';
+
+async function main() {
+  if (!IN) console.error('Error in config: `IN` not set');
+  if (!OUT) console.error('Error in config: `OUT` not set');
+  if (!IN || !OUT) return;
+  if (IN.endsWith('/')) IN = IN.substring(0, IN.length - 1);
+  if (OUT.endsWith('/')) OUT = OUT.substring(0, OUT.length - 1);
+
+  const templateDirectoryPath = `${__dirname}/../../${TEMPLATE_DIRECTORY}`;
+  const snippetDirectoryPath = `${__dirname}/../../${SNIPPET_DIRECTORY}`;
+
+  const schemasAndGenFiles = await qt.getJSONSchemasAndGenFiles(IN, LANGUAGE);
+  const allEvents: Array<[string, string, string]> = [];
+  schemasAndGenFiles.map((sg: [any, string]) => {
```

Comment:
A few more comments in this whole file would be helpful.

Reply:
Done.